test/visualize/test_visualize.py

```diff
@@ -119,6 +131,18 @@ def test_waterfall():
         # test plotting of lists
         pypesto.visualize.waterfall([result_1, result_2])
 
+    @staticmethod
+    def test_waterfall_with_nan_inf():
+        # create the necessary results
+        result_1 = create_optimization_result_nan_inf()
+        result_2 = create_optimization_result_nan_inf()
```

Comment:
Doe the test also work if one result has nans/infs, and the other one doesn't?
Maybe briefly add such a test as well... :)